.github/workflows/terraform.yml

```diff
@@ -0,0 +1,49 @@
+---
+
+###
+### Runs checks on whole repository: lint, docs, test
+###
+
+name: terraform
+on: [pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    name: "Lint"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@master
+
+      - name: "make lint"
+        run: |
+          make lint
+
+  docs:
+    runs-on: ubuntu-latest
+    name: "Terraform docs"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@master
+
+      - name: "make gen"
+        run: |
+          make gen
+          git diff --quiet || { echo "Build Changes"; git diff; git status; false; }
+
+  test:
+    runs-on: ubuntu-latest
+    name: "Terraform test"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@master
+
+      - name: "make test"
+        run: |
+          make test ARGS="-var allowed_account_id=${TF_VAR_allowed_account_id} -var role_to_assume=${TF_VAR_role_to_assume}"
+        env:
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_REGION }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          TF_VAR_allowed_account_id: ${{ secrets.allowed_account_id }}
+          TF_VAR_role_to_assume: ${{ secrets.role_to_assume }}
```

Comment:
I don't think those variable are really set in repository secrets. Are you sure you need them?